src/OpenTelemetry.Instrumentation.Http/README.md

```diff
@@ -66,7 +66,126 @@ This instrumentation can be configured to change the default behavior by using
 `HttpClientInstrumentationOptions` and
 `HttpWebRequestioninstrumentationOptions`.
 
-TODO - describe options
+### Propagator
+
+TODO
+
+### SetHttpFlavor
+
+By default, this instrumentation does not add the `http.flavor` attribute. The
+`http.flavor` attribute specifies the kind of HTTP protocol used
+(e.g., `1.1` for HTTP 1.1). The `SetHttpFlavor` option can be used to include
+the `http.flavor` attribute.
+
+The following example shows how to use `SetHttpFlavor`.
+
+```csharp
+using Sdk.CreateTracerProviderBuilder()
+    .AddHttpClientInstrumentation(
+        options => options.SetHttpFlavor = true)
+    .AddConsoleExporter()
+    .Build();
+```
+
+### Filter
+
+This instrumentation by default collects all the outgoing HTTP requests. It
+allows filtering of requests by using the `Filter` function option.
+This can be used to filter out any requests based on some condition. The Filter
+receives the request object - `HttpRequestMessage` for HttpClient or
```

Comment:
Would it be more accurate to say `HttpRequestMessage` for .NET Core and `HttpWebRequest` for .NET Framework application?

Reply:
@CodeBlanch also to share thoughts here.
I think whether `HttpRequestMessage` or `HttpWebRequest` is passed purely based on the target framework  -.net core vs .net framework, and not based on whether HttpClient or HttpWebRequest is used. In .NET Framework, even if a user is using HttpClient, I think it'd internally use HttpWebRequest, which is why I am suggesting its best to distinguish based on .NET core vs framework.

Reply:
Ah interesting, I had assumed that the `HttpClient` instrumentation was invoked even for .NET Framework. This is easy enough for me to confirm.

Reply:
No version of HttpClient for .NET Framework (which internally uses HttpWebRequest) is instrumented with diagnostic source. So Mikel did the reflection hack.

Every version of HttpClient for .NET Core (which internally contains a DiagnosticHandler by default) is instrumented with diagnosticsource.